bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -28,215 +28,203 @@ spec:
         openshift.io/component: network
         kubernetes.io/os: "linux"
     spec:
-      # Requires fairly broad permissions - ability to read all services and network functions as well
-      # as all pods.
       serviceAccountName: ovn-kubernetes-controller
       hostNetwork: true
-      hostPID: true
       priorityClassName: "system-cluster-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/ovn - ovsdb schema (selinux blocks /etc on host)
+      # /var/lib/openvswitch -> /var/lib/ovn - ovsdb data
+      # /run/openvswitch -> /run/ovn - ovsdb sockets
+      # configmap ovn-env-overrides -> /env - debug overrides
       containers:
-      # firewall rules for ovn - assumed to be setup
-      # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6641 -j ACCEPT
-      # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6642 -j ACCEPT
-      # ovs flow for ovn (geneve)
-      # /usr/share/openvswitch/scripts/ovs-ctl --protocol=udp --dport=6081 enable-protocol
-
-      # run-ovn-northd - v3
-      - name: run-ovn-northd
+      - name: northd
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "run-ovn-northd"]
-
-        volumeMounts:
-        # ovn db is stored in the pod in /etc/openvswitch (default ovs tool location)
-        # and on the host in /var/lib/openvswitch/
-        - mountPath: /etc/openvswitch/
-          name: host-var-lib-ovs
-        - mountPath: /var/lib/openvswitch/
-          name: host-var-lib-ovs
-        - mountPath: /var/run/openvswitch/
-          name: host-var-run-ovs
-
-        resources:
-          requests:
-            cpu: 100m
-            memory: 300Mi
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
```

Comment:
It seems wrong that if you want to override the master environment, you have to guess what node it's going to run on and then add node-specific overrides for that. ovnkube-node can look in `/env/${K8S_NODE}` but ovnkube-master should get its overrides from `/env/master`

Reply:
Yeah, that makes sense. I'll update it.

Reply:
I chose _master because the underscore can't appear in a domain name.